apollo-federation/src/query_plan/fetch_dependency_graph.rs

```diff
@@ -2270,6 +2270,46 @@ impl std::fmt::Display for FetchDependencyGraph {
     }
 }
 
+// Necessary for `petgraph::dot::Dot::with_attr_getters` calls to compile, but not executed.
+impl std::fmt::Display for FetchDependencyGraphNode {
+    fn fmt(&self, _f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        unimplemented!()
+    }
+}
+
+// Necessary for `petgraph::dot::Dot::with_attr_getters` calls to compile, but not executed.
+impl std::fmt::Display for FetchDependencyGraphEdge {
+    fn fmt(&self, _f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        unimplemented!()
+    }
+}
```

Comment:
I understand why this is needed, but I still dislike it. I think rather than panic here, it's probably a better idea to return an `Err`. Unfortunately, `std::fmt::Error` does not contain any data, but I think this is better than an immediate panic.